vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+#include <stdio.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
+
+#ifndef _WIN32
+#include <sys/mman.h>
+#endif
+
+using namespace std;
+typedef float weight;
+
+class weight_vector;
+
+class weights_iterator_iterator
```

Comment:
Regarding your iterator classes:

You should really look at boost::iterator and boost::range libraries.  Your iterators are not complete in the sense of C++ iterators (for example, missing the -- operators) but also not having their iterator traits defined correctly.  That means they they may not inter-operate with the C++ standard algorithms properly.  Many of the algorithms in the STL will dispatch to different implementations depending on the capabilities of iterators, meaning that your iterators below could  lead to sub-optimal code (or they may not be able to be used at all).

Also, in this case, the iterator_iterator may not really be necessary at all (see my comments further on).